### PR TITLE
View management

### PR DESCRIPTION
**Pull request by cdavidjean**
- Moved to linter 2.0
- Redid custom view management to use the atom views
- Type Hierarchy
- Plugin status
- References
- Debugger
- Run Console (bottom)

### CODE REVIEW

**Comment by cdavidjean** [comment]
Conflicts remains ? :(

**Comment by cdavidjean** [comment]
I think I have it now, was missing git add and git commit for the merged files.  I should've noticed... :)

**Comment by devoncarew** [comment]
Awesome! Looks like the PR is right with git; will review today.

File: lib/debug/debugger_ui.dart [hunk 1]
```diff
@@ -69,15 +72,15 @@ class DebuggerView extends View {
...
   final Disposables disposables = new Disposables();
 
-  DebuggerView(this.connection) {
+  DebuggerView(String id, this.connection)
+      : toolbar = div(),
+        super(id, div()) {
     // Close the debugger view when the launch is collected?
 
-    // Close the debugger view on termination.
+    // Clean the debugger view on termination.
```
Comment (devoncarew) [inline lib/debug/debugger_ui.dart]: Reset?

File: lib/analysis/find_type.dart [hunk 2]
```diff
@@ -73,7 +74,8 @@ class FindTypeHelper implements Disposable {
           navigationManager.jumpToLocation(location.file,
               location.startLine - 1, location.startColumn - 1, location.length);
         } else {
-          FindReferencesView.showView(new ReferencesSearch('Find Type', searchTerm, results: results));
+          deps[FindReferencesHelper].openView(new ReferencesSearch('Find Type',
```
Comment (devoncarew) [inline lib/analysis/find_type.dart]: not critical, but you could add a getter for this in `lib/state.dart`

File: lib/views.dart [hunk 3]
```diff
@@ -124,267 +122,93 @@ class ViewResizer extends CoreElement {
...
-  }
-
-  bool isActiveId(String viewId) {
-    return _groups.values.any((group) => group.isActiveId(viewId));
-  }
-
-  bool hasViewId(String viewId) {
-    return _groups.values.any((group) => group.hasViewId(viewId));
-  }
+// A view that is docked on atom's side docks.
```
Comment (devoncarew) [inline lib/views.dart]: use dartdoc? (`///`)

File: lib/views.dart [hunk 4]
```diff
@@ -124,267 +122,93 @@ class ViewResizer extends CoreElement {
...
-      '${name}Panel', resizer, rightPanel ? _defaultWidth : _defaultHeight);
-
-    views.onAdded.listen(_onViewAdded);
-    views.onSelectionChanged.listen(_onActiveChanged);
-    views.onRemoved.listen(_onViewRemoved);
-  }
-
-  bool get hidden => !showing;
-  bool get showing => _panel.isVisible();
+// Manages a single or multiple DockedView.
```
Comment (devoncarew) [inline lib/views.dart]: dartdoc

Comment (cdavidjean) [inline lib/views.dart]: Done

File: lib/views.dart (hunk 3 above)
Comment (cdavidjean) [inline lib/views.dart]: Done

File: lib/debug/debugger_ui.dart (hunk 1 above)
Comment (cdavidjean) [inline lib/debug/debugger_ui.dart]: Done

**Comment by cdavidjean** [comment]
Thanks!

File: lib/analysis/find_type.dart (hunk 2 above)
Comment (devoncarew) [inline lib/analysis/find_type.dart]: sgtm

**Pull request merged**